test/go_fish_test.exs

```diff
@@ -77,4 +78,24 @@ defmodule GoFishTest do
     assert Enum.find(books, fn x -> x == 2 end)
   end
 
+  test "new game" do
+    assert :new_game == GoFish.Controller.start_game([:john, :simon])
+    assert length(GoFish.Ocean.get_state()) == 52
+    assert %{books: [], hand: [], is_my_turn: true} == GoFish.Player.get_state(:john)
+    assert %{books: [], hand: [], is_my_turn: false} == GoFish.Player.get_state(:simon)
+  end
+
+  test "game over" do
+    assert :got_cards == GoFish.Player.draw_cards(:simon, 26)
+    assert :got_cards == GoFish.Player.draw_cards(:john, 26)
+    # Ocean is now empty
+    assert :no_cards_left == GoFish.Player.take_all_your(2, :john, :simon)
+    assert :no_cards_left = GoFish.Player.take_all_your(14, :simon, :john)
+    assert %{:ocean_empty => true} = GoFish.Controller.get_state()
+    # John and Simon both only have two 8s
+    assert {:got_cards, _matches} = GoFish.Player.take_all_your(8, :john, :simon)
+    # John gets the final book and the game ends
+    assert %{:players => [], :game_state => :game_over, :ocean_empty => true, :winner => {:john, 7}} = GoFish.Controller.get_state()
+  end
+
```

Comment:
Occasionally I get this error when I run the tests.
It seems like it isn't happy about how we terminate the Player?

```bash

16:02:11.304 [error] GenServer :john terminating
** (stop) bad return value: {:stop, :normal}
Last message: {:"$gen_cast", :stop}
State: %{books: [8, 9, 10, 11, 12, 13, 14], hand: [], is_my_turn: true}

16:02:11.304 [error] GenServer :simon terminating
** (stop) bad return value: {:stop, :normal}
Last message: {:"$gen_cast", :stop}
State: %{books: [2, 3, 4, 5, 6, 7], hand: [], is_my_turn: true}

16:02:11.304 [error] GenServer GoFish.Ocean terminating
** (stop) bad return value: {:stop, :normal}
Last message: {:"$gen_cast", :stop}
State: []
collected books: 2
Book made
collected books: 2
Book made


  1) test draw cards and give me all your (GoFishTest)
     test/go_fish_test.exs:23
     ** (exit) exited in: GenServer.call(:john, {:take_all_your, 3, :simon}, 5000)
         ** (EXIT) exited in: GenServer.call(:simon, {:give_all_my, 3}, 5000)
             ** (EXIT) bad return value: {:stop, :normal}
     code: case GoFish.Player.take_all_your(3, :john, :simon) do
     stacktrace:
       (elixir 1.13.4) lib/gen_server.ex:1030: GenServer.call/3
       test/go_fish_test.exs:25: (test)
```

Reply:
Could it have something to do with comment 3 here in the docs?:
https://hexdocs.pm/ex_unit/1.12/ExUnit.html#module-example

Reply:
I've tried changing it to async: false and the error still occurs :(